shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -18,12 +18,9 @@
 #include "shell/browser/unresponsive_suppressor.h"
 #include "shell/common/gin_converters/file_path_converter.h"
 #include "ui/base/glib/glib_signal.h"
+#include "ui/gtk/gtk_ui.h"
```

Comment:
This is generating a GN warning:

```
ERROR at //electron/shell/browser/ui/message_box_gtk.cc:21:11: Including a private header.
#include "ui/gtk/gtk_ui.h"
          ^--------------
This file is private to the target //ui/gtk:gtk
```

I'm not sure what is the best way to fix it. /cc @nornagon

Reply:
Wonder if we can refactor `file_dialog_gtk` to reuse https://source.chromium.org/chromium/chromium/src/+/main:ui/gtk/select_file_dialog_linux_gtk.h via `//ui/shell_dialogs` . We already started using this component in other places https://github.com/electron/electron/pull/30663

Reply:
@zcbenz, looks like there's [already a patch](https://github.com/electron/electron/blob/3da598015bccb8f9ab6833d1163c2fe2b0f2b058/patches/chromium/ui_gtk_public_header.patch) exposing "ui/gtk/gtk_util.h", so we could just tack on this header there as well?

@deepak1556, I think we'd still need to expose the header for the usage in `message_box_gtk`, don't see an upstream class we could reuse for that.

Reply:
Yeah but it is only a single utility `gtk::SetGtkTransientForAura` we are relying on from that header, maybe we can inline it. I am good with the header patching for unblocking this PR, the refactor can be explored as follow-up.

Reply:
> looks like there's [already a patch](https://github.com/electron/electron/blob/3da598015bccb8f9ab6833d1163c2fe2b0f2b058/patches/chromium/ui_gtk_public_header.patch) exposing "ui/gtk/gtk_util.h", so we could just tack on this header there as well?

Yeah adding the header to the patch would work.

Reply:
I've added the `gtk_ui.h` header to the patch which I believe resolves this thread.